lib/parser.js

```diff
@@ -362,22 +366,25 @@ module.exports = class Parser {
     if (node === undefined) {
       node = this._t.peek();
     }
-    return this._posStack && node.location ? node.location.pos : -1;
+    return this._posStack && node.location ? node.location.start : -1;
```

Comment:
Is this supposed to return a `number` or the new `{ line: number, column: number, offset: number }` you're using? With your changes, I'm not sure if `popPos`, `getPos`, or `getEnd` are correct anymore.

If the object is what you expect to return, I don't think `-1` is a good fallback and would expect `undefined` or `null` instead.

If the number is what you expect to return, I think you'll have to do a bit more work to dig the correct value out.

Also, if this could be either `null` or `undefined`, I would recommend changing the definition in the declaration file above to reflect the possibility.

Reply:
I've rebased this on #63 and fixed the type errors.